central/complianceoperator/v2/compliancemanager/manager_impl.go

```diff
@@ -251,6 +268,23 @@ func (m *managerImpl) processRequestToSensor(ctx context.Context, scanRequest *s
 	return scanRequest, nil
 }
 
+// validateScanSettingBindingsProfiles validates that the scan setting bindings does not contains any profiles that are being referenced by the scan request
+func validateScanSettingBindingsProfiles(scanSettingBindings []*storage.ComplianceOperatorScanSettingBindingV2, profiles []string, scanRequest *storage.ComplianceOperatorScanConfigurationV2, createScanRequest bool) error {
+	for _, binding := range scanSettingBindings {
+		ssbProfiles := binding.ProfileNames
+		if binding.GetName() == scanRequest.GetScanConfigName() && !createScanRequest {
+			// Skip the scan setting binding that we are updating
+			continue
+		}
+		for _, profile := range profiles {
+			if sliceutils.Find(ssbProfiles, profile) != -1 {
+				return errors.Errorf("Scan setting binding %q already exists for profile %q", binding.GetId(), profile)
```

Comment:
Are scan setting bindings also the scan configuration names? Since users interacting with CO through ACS don't really know about the concept of scan setting bindings (it's a CO specific implementation detail), I wonder if we need to phrase this in terms of the scan configuration.

If we give users a scan setting binding, how are they going to know which scan configuration already uses that profile? Or better yet, can we just tell them that in the error message directly?

Reply:
Oh - I suppose that will only work if we assume all scan setting bindings are created by ACS... Otherwise we'd need to check if the scan setting binding belongs to an existing scan configuration. If not, then we can assume it was created as a one-off scan setting binding on that cluster.

Reply:
I think this check was for checking to make sure no existing SSB contains any profiles being used by this scan request.